http/logical.go

```diff
@@ -11,7 +11,9 @@ import (
 	"github.com/hashicorp/vault/vault"
 )
 
-func handleLogical(core *vault.Core, dataOnly bool) http.Handler {
+type PrepareRequest func(req *logical.Request) error
```

Comment:
It's not totally conventional throughout Vault but it's often nice to name types that are functions with Func at the end, so that it's obvious when you're looking at types in documentation. Can you `s/PrepareRequest/PrepareRequestFunc'?